src/commands/generateProject.ts

```diff
@@ -100,41 +96,44 @@ export async function generateProject(): Promise<void> {
       () => util.downloadFile(requestOptions, zipPath)
     );
 
+    const targetDirFolder = path.join(targetDirString, artifactId);
+
     try {
-      const targetDirFolder = path.join(targetDirString, artifactId);
       await util.unzipFile(zipPath, targetDirString, targetDirFolder);
-      try {
-        await util.deleteFile(zipPath);
-      } catch (e) {
-        console.error(e);
-        vscode.window.showErrorMessage(`Failed to delete file ${zipName}`);
-      }
-
-      // open the unzipped folder in a new VS Code window
-      const uriPath = vscode.Uri.file(targetDirFolder);
-      // prompt user whether they want to add project to current workspace or open in a new window
-      const selection = await vscode.window.showInformationMessage(
-        "MicroProfile Starter project generated.  Would you like to add your project to the current workspace or open it in a new window?",
-        ...[
-          OPEN_NEW_PROJECT_OPTIONS.ADD_CURRENT_WORKSPACE,
-          OPEN_NEW_PROJECT_OPTIONS.OPEN_NEW_WINDOW,
-        ]
-      );
-      if (selection === OPEN_NEW_PROJECT_OPTIONS.ADD_CURRENT_WORKSPACE) {
-        vscode.workspace.updateWorkspaceFolders(0, 0, { uri: uriPath });
-      } else if (selection === OPEN_NEW_PROJECT_OPTIONS.OPEN_NEW_WINDOW) {
-        await vscode.commands.executeCommand("vscode.openFolder", uriPath, true);
-      }
-    } catch (err) {
-      console.error(err);
-      vscode.window.showErrorMessage("Failed to extract the MicroProfile Starter project.");
+    } catch (e) {
+      console.error(e);
+      const err = new Error("Unable to extract MicroProfile Starter Project");
+      err.name = ERRORS.EXTRACT_PROJECT_ERROR;
+      throw err;
+    }
+
+    // if failed to delete the zip, no need to error out but show a warning to users
+    try {
+      await util.deleteFile(zipPath);
+    } catch (e) {
+      console.error(e);
+      vscode.window.showErrorMessage(`Failed to delete file ${zipName}`);
+    }
+
+    const uriPath = vscode.Uri.file(targetDirFolder);
+    // prompt user whether they want to add project to current workspace or open in a new window
+    const selection = await vscode.window.showInformationMessage(
+      "MicroProfile Starter project generated.  Would you like tPo add your project to the current workspace or open it in a new window?",
```

Comment:
Typo `Would you like tPo` -> `Would you like to`

Reply:
```suggestion
      "MicroProfile Starter project generated.  Add project to the current workspace or open it in a new window?",
```